examples/cmd/logtransport.go

```diff
@@ -22,24 +23,31 @@ type logTransport struct {
 func (t *logTransport) RoundTrip(r *http.Request) (*http.Response, error) {
 	t0 := time.Now()
 
-	reqBody, _ := ioutil.ReadAll(r.Body)
-	r.Body = ioutil.NopCloser(bytes.NewBuffer(reqBody))
+	backupReqBody, err := httpbody.BackupRequestBody(r)
+	if err != nil {
+		return nil, fmt.Errorf("could not backup the request before sending it: %s", err)
+	}
 
 	resp, err := t.T.RoundTrip(r)
 	if err != nil {
-		return resp, err
+		return nil, err
 	}
 
-	respBody, err := ioutil.ReadAll(resp.Body)
+	reqBody, err := ioutil.ReadAll(backupReqBody)
 	if err != nil {
-		return resp, err
+		t.Logger.With(
+			log.Fields{"elapsed": time.Since(t0), "response-code": resp.StatusCode, "url": r.URL},
+		).Warningf("could not log the request: %s", err)
+	}
+
+	respBody, err := httpbody.ReadResponseAndRestore(resp)
```

Comment:
why for req it's named `BackupRequestBody` but for resp `ReadResponseAndRestore`? They do absolutely the same thing, no?

Reply:
we already talked about this.
They both do different things and have different purposes.
https://github.com/src-d/metadata-retrieval/pull/57#discussion_r335351660